plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraSplitManager.java

```diff
@@ -0,0 +1,107 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestCassandraSplitManager
+{
+    private static final String KEYSPACE = "test_cassandra_split_manager_keyspace";
+
+    private CassandraServer server;
+    private CassandraSession session;
+
+    @BeforeAll
+    public void setUp()
+            throws Exception
+    {
+        server = new CassandraServer();
+        session = server.getSession();
+        createKeyspace(session, KEYSPACE);
+    }
+
+    @AfterAll
+    public void tearDown()
+    {
+        server.close();
+        server = null;
+        session.close();
+        session = null;
+    }
+
+    @Test
+    public void testGetSplitsWithSinglePartitionKeyColumn()
+            throws Exception
+    {
+        String tableName = "single_partition_key_column_table";
+        int partitionCount = 3;
+
+        session.execute(format("""
+            CREATE TABLE %s.%s (
+              partition_key int,
+              clustering_key text,
+              PRIMARY KEY(partition_key, clustering_key)
+            )
+        """, KEYSPACE, tableName));
+
+        CassandraColumnHandle columnHandle = new CassandraColumnHandle("partition_key", 0, CassandraTypes.INT, true, false, false, false);
+        ImmutableList.Builder<CassandraPartition> partitions = ImmutableList.builder();
+        for (int i = 0; i < partitionCount; i++) {
+            TupleDomain<ColumnHandle> tupleDomain = TupleDomain.fromFixedValues(Map.of(columnHandle, NullableValue.of(CassandraTypes.INT.getTrinoType(), (long) i)));
+            partitions.add(new CassandraPartition(new byte[] {0, 0, 0, (byte) i}, format("\"partition_key\" = %d", i), tupleDomain, false));
+            session.execute(format("INSERT INTO %s.%s (partition_key, clustering_key) VALUES (%d, '%d')", KEYSPACE, tableName, i, i));
+        }
+
+        CassandraTableHandle tableHandle = new CassandraTableHandle(
+                new CassandraNamedRelationHandle(KEYSPACE, tableName, Optional.of(partitions.build()), ""));
+        CassandraSplitManager splitManager = newSplitManager(2, null);
+        try (ConnectorSplitSource splitSource = splitManager.getSplits(null, null, tableHandle, null, null)) {
+            List<ConnectorSplit> splits = splitSource.getNextBatch(100).get().getSplits();
+            assertThat(splits).hasSize(2);
+            assertThat(((CassandraSplit) splits.get(0)).getPartitionId()).isEqualTo("\"partition_key\" in (0,1)");
+            assertThat(((CassandraSplit) splits.get(1)).getPartitionId()).isEqualTo("\"partition_key\" in (2)");
+        }
+
+        session.execute(format("DROP TABLE %s.%s", KEYSPACE, tableName));
+    }
+
+    private CassandraSplitManager newSplitManager(int partitionSizeForBatchSelect, CassandraTokenSplitManager tokenSplitManager)
```

Comment:
Let's inline this method as it's not reused.

Reply:
I've removed the method in the commit https://github.com/trinodb/trino/pull/21940/commits/006f2dd3ad8115a922d15d7a44d1199a05069040.
It might be better to add the same method again when we add another test.